### PR TITLE
Feature/timeout - Add timeout support

### PR DESCRIPTION
**Pull request by spike008t**
Hello, as you asked in my commit 937dcddef9913cd7bcec9a4c3ba93ef65c83f96e.

Here the PR. I've also add the support of an error.

Hope that is ok.

### CODE REVIEW

**Comment by doublerebel** [comment]
Excellent, thanks!  Would you be willing to write a test for this feature?

**Comment by spike008t** [comment]
Yeah no problem
Le 4 févr. 2015 21:07, "Charles Phillips" <notifications@github.com> a
écrit :

File: lib/najax.js [hunk 1]
```diff
@@ -213,6 +213,22 @@ function request(options, a, b) {
...
+		req.on('socket', function(socket) {
+			socket.setTimeout(o.timeout);
+			socket.on('timeout', function() {
+				req.abort();
+
+				if (_.isFunction(o.error)) {
+					o.error(jqXHR, 'timeout');
+				}
+
+				dfd.reject(jqHXR, 'timeout');
```
Comment (wololodev) [inline lib/najax.js]: varible name should be jqXHR instead of jqHXR lol

Comment (alanclarke) [inline lib/najax.js]: lol! @wololodev @spike008t if you make the change we'll merge

Comment (spike008t) [inline lib/najax.js]: @wololodev @alanclarke yeah lol sorry for that!

I'll send an other PR

**Comment by spike008t** [comment]
Updated.

Otherwise I cannot test the timeout socket, couldn't find the right way to do that.

**Comment by alanclarke** [comment]
hey @spike008t I wrote a test for you, could you add it to your pr?

see here: https://github.com/najaxjs/najax/commit/19d5728ff3da59ae1d98efe303dcc97fe2874ea8

**Comment by spike008t** [comment]
Done!

**Pull request merged**